### PR TITLE
Update react 0.14.2

### PR DESCRIPTION
**Pull request by jcgertig**

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -103,6 +103,14 @@ module.exports = yeoman.generators.Base.extend({
         this.templatePath('_server.js'),
         this.destinationPath('server.js')
       );
+      this.fs.copy(
+        this.templatePath('_Procfile'),
+        this.destinationPath('Procfile')
+      );
```
Comment (iamdustan) [inline app/index.js]: Procfile, really?

File: app/templates/_package.json [hunk 2]
```diff
@@ -6,54 +6,58 @@
...
-    "react-tween-state": "0.0.5",
-    "source-map-support": "^0.2.10",
-    "suitcss-preprocessor": "^0.4.0"
+    "normalize.css": "^3.0.3",
+    "object.assign": "^3.0.0",
+    "postcss-import": "^7.1.0",
+    "postcss-loader": "^0.7.0",
+    "precss": "^1.3.0",
+    "react": "^0.14.3",
+    "react-cookie": "^0.3.4",
```
Comment (iamdustan) [inline app/templates/_package.json]: Idk if this should be standard in here.

File: app/templates/scripts/_webpack.base.js [hunk 3]
```diff
@@ -118,29 +123,37 @@ module.exports = function(options) {
...
 
     devtool: 'sourcemap',
 
-    rework: {
-      use: [
-        reworkLoader.plugins.imports,
-        reworkLoader.plugins.urls,
-      ],
+    postcss: function (webpack) {
+        return [
```
Comment (iamdustan) [inline app/templates/scripts/_webpack.base.js]: tab index is off (2 space)

File: app/templates/scripts/_webpack.base.js [hunk 4]
```diff
@@ -118,29 +123,37 @@ module.exports = function(options) {
...
+            postcssImport({
+                addDependencyTo: webpack
+            }),
+            precss({
```
Comment (iamdustan) [inline app/templates/scripts/_webpack.base.js]: what’s `precss`?

File: component/templates/index.js [hunk 5]
```diff
@@ -2,10 +2,14 @@
 
 require('./styles.css');
 
-import React from 'react';
-var {PropTypes} = React;
+import React, {Component, PropTypes} from 'react';
+
+class <%= ComponentName %> extends Component {
+
+  state = {
```
Comment (iamdustan) [inline component/templates/index.js]: rm this. I’d rather not stick state into every component by default.

File: handler/templates/handler.js [hunk 6]
```diff
@@ -1,28 +1,29 @@
-import React from 'react';
-import {Resolver} from 'react-resolver';
+/** @flow */
+
+require('./styles.css');
```
Comment (iamdustan) [inline handler/templates/handler.js]: you should totally add css modules in here (with the other changes).

`import styles from './styles.css'`

and then `className={styles['<%= RouteName%>']}` down below

File: handler/templates/handler.js [hunk 7]
```diff
@@ -1,28 +1,29 @@
...
+
+import React, {Component, PropTypes} from 'react';
+
+class <%= RouteName %> extends Component {
+
+  state = {
+    msg: 'Hello World'
+  };
```
Comment (iamdustan) [inline handler/templates/handler.js]: no state!

File: package.json [hunk 8]
```diff
@@ -24,6 +24,8 @@
   ],
   "dependencies": {
     "chalk": "^1.0.0",
+    "font-awesome": "^4.3.0",
+    "font-awesome-webpack": "0.0.3",
```
Comment (iamdustan) [inline package.json]: not out of the gate, por favor.

**Comment by iamdustan** [comment]
react 0.14.3 is out. ;)

CI is failing.

Seriously thank you for upgrading this!

**Comment by jcgertig** [comment]
just the name its up to 0.14.3

File: package.json (hunk 8 above)
Comment (jcgertig) [inline package.json]: lol I ment to remove that

File: handler/templates/handler.js (hunk 7 above)
Comment (jcgertig) [inline handler/templates/handler.js]: this was to show an example of how state should be done with es7 and 0.14.3

File: handler/templates/handler.js (hunk 6 above)
Comment (jcgertig) [inline handler/templates/handler.js]: maybe

File: app/templates/scripts/_webpack.base.js (hunk 4 above)
Comment (jcgertig) [inline app/templates/scripts/_webpack.base.js]: https://jonathantneal.github.io/precss/

File: app/templates/_package.json (hunk 2 above)
Comment (jcgertig) [inline app/templates/_package.json]: true

File: app/index.js (hunk 1 above)
Comment (jcgertig) [inline app/index.js]: yes lol

**Comment by jcgertig** [comment]
I have no control over the ci

**Comment by jcgertig** [comment]
don't see a reason to test iojs and node 0.12 should test with 4.4.4 and up

File: handler/templates/handler.js (hunk 7 above)
Comment (iamdustan) [inline handler/templates/handler.js]: I don’t :heart: static property initializers being an expected thing for users to use by using this yet...

File: app/templates/scripts/_webpack.base.js (hunk 4 above)
Comment (iamdustan) [inline app/templates/scripts/_webpack.base.js]: is that only being used for vars?

Why not use something like https://github.com/MadLittleMods/postcss-css-variables to get the real CSS variable syntax?

File: handler/templates/handler.js (hunk 7 above)
Comment (jcgertig) [inline handler/templates/handler.js]: why not?

File: app/templates/scripts/_webpack.base.js (hunk 4 above)
Comment (jcgertig) [inline app/templates/scripts/_webpack.base.js]: it also allows for loops and extends

Comment (jcgertig) [inline app/templates/scripts/_webpack.base.js]: but I do like postcss-css-variables  had not looked into it

File: handler/templates/handler.js (hunk 6 above)
Comment (jcgertig) [inline handler/templates/handler.js]: dont really like the scoping stuff with it

Comment (jcgertig) [inline handler/templates/handler.js]: will require more reading

File: package.json (hunk 8 above)
Comment (jcgertig) [inline package.json]: was in the wrong folder when I ran the npm i for those :P

**Comment by jcgertig** [comment]
@iamdustan could you also look at https://github.com/jcgertig/rn-router

**Comment by jcgertig** [comment]
@iamdustan some of the suggested changes

**Pull request merged**